scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/XmlParser.scala

```diff
@@ -0,0 +1,155 @@
+package scala.meta.internal.tokenizers
+
+import scala.annotation.switch
+import scala.meta.Dialect
+import scala.meta.inputs.Input
+
+import fastparse.all._
+
+/**
+  * Copy-pasta from this lihaoyi comment:
+  * [[https://github.com/scalameta/fastparse/pull/1#issuecomment-244940542]]
+  * and adapted to more closely match scala-xml
```

Comment:
@lihaoyi Can we include your code in our repository under the Scala License? (https://github.com/scalameta/scalameta/blob/master/LICENSE.md)